lite/src/lib.rs

```diff
@@ -0,0 +1,428 @@
+            .task_center
+            .address_book()
+            .expect("address book is always set at creation");
+        push_addresses(
+            address_book.get_bound_addresses::<HttpIngressPort>(),
+            &mut addresses,
+        );
+
+        push_addresses(
+            address_book.get_bound_addresses::<AdminPort>(),
+            &mut addresses,
+        );
+
+        push_addresses(
+            address_book.get_bound_addresses::<FabricPort>(),
+            &mut addresses,
+        );
+
+        push_addresses(
+            address_book.get_bound_addresses::<ControlPort>(),
+            &mut addresses,
+        );
+
+        addresses
+    }
+
+    pub fn get_advertised_addresses(&self) -> Vec<AddressMeta> {
+        let mut addresses = Vec::with_capacity(3);
+        let address_book = self
+            .task_center
+            .address_book()
+            .expect("address book is always set at creation");
+
+        if self.config.has_role(Role::HttpIngress) {
+            push_advertised(
+                self.config.ingress.advertised_address(address_book),
+                &mut addresses,
+            );
+        }
+
+        if self.config.has_role(Role::Admin) {
+            push_advertised(
+                self.config.admin.advertised_address(address_book),
+                &mut addresses,
+            );
+        }
+
+        // fabric
+        push_advertised(
+            self.config.common.advertised_address(address_book),
+            &mut addresses,
+        );
+
+        addresses
+    }
+
+    pub async fn create(opts: Options) -> Result<Restate> {
+        if rlimit::increase_nofile_limit(u64::MAX).is_err() {
+            warn!("Failed to increase the number of open file descriptors limit.");
+        }
+
+        let listener_options = ListenerOptionsBuilder::default()
+            .use_random_ports(Some(opts.use_random_ports))
+            .listen_mode(Some(if opts.enable_tcp {
+                ListenMode::All
+            } else {
+                ListenMode::Unix
+            }))
+            .build()
+            .unwrap();
+
+        let mut common_builder = CommonOptionsBuilder::default();
+
+        common_builder
+            .roles(Role::Worker | Role::HttpIngress | Role::MetadataServer | Role::Admin)
+            .rocksdb_total_memory_size(opts.memory_budget)
+            .node_name(Some("embedded".to_owned()))
+            .force_node_id(Some(PlainNodeId::new(1)))
+            .disable_telemetry(true)
+            .fabric_listener_options(listener_options)
+            .default_num_partitions(1)
+            .default_replication(ReplicationProperty::new(NonZero::new(1).unwrap()))
+            .disable_prometheus(true);
+
+        if let Some(data_dir) = opts.data_dir {
+            common_builder.base_dir(data_dir);
+        }
+
+        let common = common_builder.build()?;
+
+        let bifrost = BifrostOptionsBuilder::default()
+            .default_provider(ProviderKind::Local)
```

Comment:
Didn't we want to remove the local provider kind at some point to reduce the maintenance surface?